### DISCUSSION TITLE
Publish weather data with MQTT to Thingsboard

### DISCUSSION THREAD

**Discussion by jonasijungelen** [merbanan/rtl_433#2880] [Q&A]
Hi,

Newbie here. Is it possible to publish mqtt data directly to Thingsboard with the right command using rtl_433? I dont understand quite yet how an mqtt broker works. I can read the weather data from a cheap weather station perfectly fine, and now i want to send it to a cloud service so that I can visualize the data with some nice widgets.

I have tried this code, but doesn work:

`rtl_433 -R 215 -F  "mqtt://mqtt.thingsboard.cloud:1883,user=USER,pass=PASS"`

Screenshot from Thingsboard device setup.

<img width="1398" alt="Skjermbilde 2024-03-19 kl  20 33 26" src="https://github.com/merbanan/rtl_433/assets/82219250/502e6412-44eb-47cc-85fc-ded283303ccd">

**Comment by zuckschwerdt** [comment]
The command looks ok. There is no option to manually set a client ID though, is that required to work with the service?

**Reply by jonasijungelen** [reply]
I´m not sure it is required. Will try look into it.

**Reply by jonasijungelen** [reply]
just decided to ditch Thingsboard and go for InfluxDB and Grafana. Do I need to run a mqtt broker service on the raspberry or can rtl_433 send mqtt directly influxdb cloud?

**Reply by zuckschwerdt** [reply]
Good choice. InfluxDB is directly supported. S.a. `rtl_433 -F`
